spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -245,24 +245,25 @@ private String mapToFieldName(PropertyDescriptor propertyDescriptor) {
 	}
 
 	private Slice executeSliceQuery(Object[] parameters) {
+		DatastoreNextPageAwareResultsIterable<?> results =
+				(DatastoreNextPageAwareResultsIterable<?>)
+						this.datastoreOperations.queryKeysOrEntities(buildSliceQuey(parameters), this.entityType);
+
+		return (Slice) this.processRawObjectForProjection(
+				new SliceImpl(results.toList(),
+						getPageable(parameters, results.getCursor()),
+						results.hasNextPage()));
```

Comment:
Does it make sense to make `SliceImpl` also evaluate `hasNext()` lazily?

Reply:
It is a Spring Data class and expects a boolean parameter.

Reply:
Can we extend `SliceImpl`?